services/src/main/java/org/exoplatform/addons/gamification/rest/AnnouncementRest.java

```diff
@@ -88,11 +100,24 @@ public Response getAllAnnouncementByChallenge(@ApiParam(value = "id of the chall
     if (limit <= 0) {
       return Response.status(Response.Status.BAD_REQUEST).entity("Limit must be positive").build();
     }
+    EntityTag eTag = null ;
+    List<AnnouncementRestEntity> announcementsRestEntities = new ArrayList<>();
+    eTag = new EntityTag(String.valueOf(announcementsRestEntities.hashCode()));
```

Comment:
unneeded, you have to calculate Etag once you have loaded the entities to cache the response.

Reply:
done!